client/components/site-selector-modal/index.jsx

```diff
@@ -35,45 +37,46 @@ class SiteSelectorModal extends Component {
 		// className: class name(s) to be added to the Dialog
 		className: PropTypes.string,
 		// from localize()
-		translate: PropTypes.func.isRequired
+		translate: PropTypes.func.isRequired,
+		// connected props
+		primarySiteId: PropTypes.number
 	}
 
 	constructor( props ) {
 		super( props );
 
-		const primarySite = sitesList.getPrimary();
-		let filteredSites = sitesList.getVisible();
+		const { primarySiteId } = this.props;
+		let filteredSiteIds = sitesList.getVisible().map( ( site ) => site.ID );
 
 		if ( props.filter ) {
-			filteredSites = filteredSites.filter( props.filter );
+			filteredSiteIds = filteredSiteIds.filter( props.filter );
 		}
 
 		this.state = {
-			site: includes( filteredSites, primarySite )
-				? primarySite
-				: filteredSites[ 0 ]
+			siteId: includes( filteredSiteIds, primarySiteId )
```

Comment:
Can do, once we obtain visible sites from Redux instead of `sites-list`. Curious -- do you mean just adding a `getVisibleSiteIds` one, or one that actually checks if `primarySite` is part of that list? That'd need to accept a `filter` function prop and would be rather specific -- something like `doFilteredSitesIncludePrimarySite( state, filter )`?. Or am I reading you totally wrong?